### ISSUE TITLE
compilation and first plot extremely slow

### ISSUE THREAD

**Issue by stakaz** [JuliaPlots/Plots.jl#2211]
Hello, I found some other issues dealing with this problem, but my numbers are completely out of scope of anything else I have seen so far.

On 1.3.0-rc2 I get the following results for the start of Plots and call plot for the first time:

```julia
julia> @time using Plots
 36.457346 seconds (11.66 M allocations: 601.518 MiB, 2.04% gc time)

julia> @time gr()
  0.206046 seconds (46.11 k allocations: 2.269 MiB)
Plots.GRBackend()

julia> @time plot([1,2,3], [4,5,6])
 87.884694 seconds (46.10 M allocations: 2.259 GiB, 4.72% gc time)
```
I have similar results with julia 1.1.0.
Is this the current "normal" behavior or is it way to slow?

**Comment by mkborregaard** [comment]
That's way off the charts

**Comment by mkborregaard** [comment]
Here are my numbers on 1.3.0-rc2:
```
julia> @time using Plots
  7.627296 seconds (10.42 M allocations: 550.638 MiB, 5.06% gc time)

julia> @time gr()
  0.010706 seconds (14.87 k allocations: 818.267 KiB)
Plots.GRBackend()

julia> @time plot([1,2,3], [4,5,6])
 18.717534 seconds (43.56 M allocations: 2.134 GiB, 6.98% gc time)
```

what's your `versioninfo()`?

**Comment by stakaz** [comment]
```julia
julia> versioninfo()
Julia Version 1.3.0-rc2.0
Commit a04936e3e0 (2019-09-12 19:49 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Pentium(R) CPU  N3530  @ 2.16GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, silvermont)
Environment:
  JULIA_NUM_THREADS = 3
  JULIA_EDITOR = "/usr/lib/electron4/electron"
```

**Comment by mkborregaard** [comment]
That's quite weird, that system isn't even that old. Do you have long compilation times on other packages as well, or just Plots?

**Comment by stakaz** [comment]
Well, I am not sure, here are two examples:

```julia
julia> @time using DataFrames
  3.950280 seconds (1.50 M allocations: 93.240 MiB, 0.62% gc time)

julia> @time using PyPlot
 33.090317 seconds (16.30 M allocations: 855.425 MiB, 3.46% gc time)

```

**Comment by mkborregaard** [comment]
DataFrames is 1.8 secs on mind. But your processor is also a lot slower. A bit unsure... Maybe open a question on Discourse rather than here?

**Comment by Sov-trotter** [comment]
This is dead slow. Here are my benchmarks: 
Julia Version 1.2.0
Commit c6da87ff4b (2019-08-20 00:03 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)

julia > @time using Plots
  3.482907 seconds (4.51 M allocations: 259.478 MiB, 6.65% gc time)

**Comment by stakaz** [comment]
Yeah, thats true. Hmm, do you have some specific flags for julia to start or a setup.jl?

**Comment by stakaz** [comment]
On a freshly downloaded 1.2.0 version (only installed and precompiled Plots):

```julia
julia> versioninfo()
Julia Version 1.2.0
Commit c6da87ff4b (2019-08-20 00:03 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Pentium(R) CPU  N3530  @ 2.16GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, silvermont)

julia> @time using Plots
 15.629353 seconds (4.46 M allocations: 257.699 MiB, 3.77% gc time)

julia> @time gr()
  0.197046 seconds (81.34 k allocations: 4.045 MiB, 5.59% gc time)
Plots.GRBackend()

julia> @time plot([1,2,3], [4,5,6])
 73.604810 seconds (46.15 M allocations: 2.272 GiB, 5.15% gc time)

``` 

Better, but still bad.

**Comment by stakaz** [comment]
And for PyPlot:

```julia
julia> @time using PyPlot
 18.028322 seconds (7.76 M allocations: 405.917 MiB, 2.92% gc time)
```

**Comment by stakaz** [comment]
I found one point which was different for 1.2.0 and 1.3.0 calls. I have used `-O3` in the 1.3.0 call.

Now, when removing it I get ≈16s result also in the case of 1.3.0 version.

It is still slow but two times faster than before.

**Comment by Sov-trotter** [comment]
Nope, no flags of any sort. Only once I started like, `julia -p 8` which is the nworkers flag, but that only ended up doubling over the compile times. 
Also you seem to be using a quite slow CPU so the times are self-explanatory.
![Screenshot from 2019-10-10 14-18-25](https://user-images.githubusercontent.com/43717431/66553834-80697880-eb69-11e9-96af-717829cfb8a5.png)
Above you can see the ss from conky, showing the CPU usage during those long compile times.
PS : I have a base clock of 1.60GHz.

**Comment by stakaz** [comment]
Is the 2.16GHz not the base clock?

**Comment by Sov-trotter** [comment]
CPU: Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
And that's what hurts me. Like for every compile operation it has to enable turbo boost. ;-;

**Comment by BeastyBlacksmith** [comment]
https://github.com/JuliaPlots/Plots.jl/pull/2544 should help here, otherwise there is not much we can do
